postgis/lwgeom_functions_basic.c

```diff
@@ -2064,6 +2064,69 @@ Datum ST_MakeEnvelope(PG_FUNCTION_ARGS)
 	PG_RETURN_POINTER(result);
 }
 
+
+PG_FUNCTION_INFO_V1(ST_TileEnvelope);
+Datum ST_TileEnvelope(PG_FUNCTION_ARGS)
+{
+	LWPOLY *poly;
+	GSERIALIZED *result;
+	uint32_t x, y, zoom;
+	int32_t worldTileSize;
+	double worldGeoSize, tileGeoSize;
+	double boundsWidth, boundsHeight;
+	double x1, y1, x2, y2;
+	int32_t srid = 3857;
+	GBOX *bounds;
+	GBOX bounds_3857;
+
+	POSTGIS_DEBUG(2, "ST_TileEnvelope called");
+
+	zoom = PG_GETARG_INT32(0);
+	x = PG_GETARG_INT32(1);
+	y = PG_GETARG_INT32(2);
+	if (PG_NARGS() > 3)
+		srid = PG_GETARG_INT32(3);
+	if (PG_NARGS() > 4)
+		bounds = (GBOX*)(PG_GETARG_POINTER(4));
+	else
+	{
+		gbox_init(&bounds_3857);
+		bounds_3857.ymin = bounds_3857.xmin = -20037508.3427892;
```

Comment:
do we have any kind of get_box_by_srid function? it's a thing I'd need for #436 too.

Reply:
@roualt is there any concept of "projection boundaries" so we can ask these numbers from Proj? I see these values on some websites but don't know how are these collected.

Reply:
It's part of the projection definition. It comes from the Earth equatorial radius (defined by WGS-84 
 as `6378137`) times 2*PI to get the circumference. But I don't know if we can get it "dinamically", but I'd try to avoid it as creating projections is expensive with PROJ6+

Reply:
> is there any concept of "projection boundaries" so we can ask these numbers from Proj? I see these values on some websites but don't know how are these collected.

yes, proj_get_area_of_use() exists. It returns values in lat / long, and in the case of EPSG:3857, they use latitude = 85.06 deg, which translates to northing = 20048966.1040146. So those aren't appropriate for the well-defined GoogleMaps tiling scheme.

Reply:
Unfortunately the projection area of validity != area of tiling, even for a "simple" case like web mercator, let alone an arbitrary CRS. I feel like this utility really is constrained to handling the web mercator case.